### PR TITLE
describe changes req. to create debian packages

### PR DESCRIPTION
**Pull request by domdorn**
describe whats changed with the sbt-native-packager / debian package creation. 
tracked in bug report #4329

### CODE REVIEW

File: documentation/manual/releases/Migration24.md [hunk 1]
```diff
@@ -512,6 +512,19 @@ Previously, Play added all the resources to the the `conf` directory in the dist
...
+We've upgraded the version of the [sbt-native-packager](https://github.com/sbt/sbt-native-packager). Due to this, the following adjustments might be necessary:
+ * Change of syntax of the `/etc/default/$appname` file: The syntax of this file has changed. In previous versions of the Play-Framework this file allowed you to directly set additional command line parameters. The new syntax of the file allows you to set various environment variables.
+ * The equivalent to the old syntax of the default file is an `application.ini` file in your archive's `conf` folder.
+ * The default-file gets sourced by `SystemV` Init scripts only (this might be a bug) -  Upstart ignores this file right now. To change your build to create `SystemV` compatible packages, add this to your build.sbt: 
+```
+import com.typesafe.sbt.packager.archetypes.ServerLoader.{SystemV, Upstart}
+
+serverLoading in Debian := SystemV
+```
+ * Other changes that might be necessary can be found in the [sbt-native-packager release notes](https://github.com/sbt/sbt-native-packager/releases). [This bug report](https://github.com/playframework/playframework/issues/4329) includes the problems we've found with the upgrade.
```
Comment (jroper) [inline documentation/manual/releases/Migration24.md]: I don't think referencing the bug report from the migration guide is appropriate, it's like saying "here's a bug that we couldn't be bothered fixing so go and look at it yourself".  If there is any useful information in the bug report that is needed in the migration guide, then it should be in the migration guide.

File: documentation/manual/releases/Migration24.md [hunk 2]
```diff
@@ -512,6 +512,19 @@ Previously, Play added all the resources to the the `conf` directory in the dist
 
 This can be turned off by setting `PlayKeys.externalizeResources := false`, which will cause no `conf` directory to be created in the distribution, and it will not be on the classpath.  The contents of the applications `conf` directory will still be on the classpath by virtue of the fact that it's included in the applications jar file.
 
+### Changes in Debian Package creation
+
+We've upgraded the version of the [sbt-native-packager](https://github.com/sbt/sbt-native-packager). Due to this, the following adjustments might be necessary:
```
Comment (jroper) [inline documentation/manual/releases/Migration24.md]: Generally, we try to avoid second person - just say "The sbt-native-packager has been upgraded"

File: documentation/manual/releases/Migration24.md [hunk 3]
```diff
@@ -512,6 +512,19 @@ Previously, Play added all the resources to the the `conf` directory in the dist
...
+ * Change of syntax of the `/etc/default/$appname` file: The syntax of this file has changed. In previous versions of the Play-Framework this file allowed you to directly set additional command line parameters. The new syntax of the file allows you to set various environment variables.
```
Comment (jroper) [inline documentation/manual/releases/Migration24.md]: This is a bit wordy, maybe:

> The syntax of `/etc/default/$appname` has changed from being a simple list of additional command line parameters to being a shell script that gets sourced by the start script, allowing you to set environment variables.

File: documentation/manual/releases/Migration24.md [hunk 4]
```diff
@@ -512,6 +512,19 @@ Previously, Play added all the resources to the the `conf` directory in the dist
...
+ * The equivalent to the old syntax of the default file is an `application.ini` file in your archive's `conf` folder.
+ * The default-file gets sourced by `SystemV` Init scripts only (this might be a bug) -  Upstart ignores this file right now. To change your build to create `SystemV` compatible packages, add this to your build.sbt:
```
Comment (jroper) [inline documentation/manual/releases/Migration24.md]: If it's a bug, it should be reported on sbt-native-packager, and possibly fixed in a future version of Play. The migration guide is not the right place to flag bugs, remove the bit saying this might be a bug.

File: documentation/manual/releases/Migration24.md [hunk 5]
```diff
@@ -512,6 +512,19 @@ Previously, Play added all the resources to the the `conf` directory in the dist
...
+The [sbt-native-packager](https://github.com/sbt/sbt-native-packager) has been upgraded. Due to this, the following adjustments might be necessary:
+ * The syntax of the `/etc/default/$appname` file has changed from being a simple list of command line parameters to being a shell script that gets sourced by the start/stop scripts, allowing you to set environment variables.
+ * The equivalent to the old syntax of the default file is an `application.ini` file in your archive's `conf` folder.
+ * The default-file gets sourced by `SystemV` Init scripts only - Upstart ignores this file right now. To change your build to create `SystemV` compatible packages, add this to your build.sbt: 
+```
+import com.typesafe.sbt.packager.archetypes.ServerLoader.{SystemV, Upstart}
+
+serverLoading in Debian := SystemV
+```
+ * Other changes that might be necessary can be found in the [sbt-native-packager release notes](https://github.com/sbt/sbt-native-packager/releases).
```
Comment (marcospereira) [inline documentation/manual/releases/Migration24.md]: @domdorn should we link to the specific release of sbt-native-packager (like https://github.com/sbt/sbt-native-packager/releases/tag/v1.0.0) instead of a page listing all releases?

**Comment by benmccann** [comment]
We usually ask that you squash your commits before we merge them to keep the commit history clean. (`git rebase -i HEAD~2` & `git push -f`)

**Comment by domdorn** [comment]
I try to do that tomorrow.. I made the doc change online using the github editor as I was on a slow connection and couldn't download the ~100 mb of the play source... unfortunately github does not allow me to squash the commits online.

**Comment by domdorn** [comment]
i squashed them.
